src/schema/index.js

```diff
@@ -10,6 +10,7 @@ const dateTimeSchema = Joi.array().min(3).max(7).ordered(
 )
 
 const durationSchema = Joi.object().keys({
+  before: Joi.boolean(),//option to set before alaram
```

Comment:
Rather than adding a `before` property, perhaps we can let the user pass in a negative duration, e.g. `{ hours: -1 }`

Reply:
for that the syntax is TRIGGER:-PT6H30M, so its good to keep before ?